log4j-api/src/main/java/org/apache/logging/log4j/util/SortedArrayStringMap.java

```diff
@@ -128,6 +128,21 @@
     private boolean immutable;
     private transient boolean iterating;
 
+    /**
+     * Transforms a generic map to a {@link StringMap} by mapping the keys to
+     * {@link String}.
+     * 
+     * @param map a generic map
+     * @return a {@link StringMap}
+     */
+    public static SortedArrayStringMap fromGenericMap(final Map<?, ?> map) {
```

Comment:
I don't think this is needed, would you check your tests against HEAD, without any main changes?

Reply:
@garydgregory: after your fix it is not needed anymore.